### PR TITLE
Build Git from source code

### PR DESCRIPTION
**Pull request by pascalberger**
Builds Git from source code, which allows us to keep Git version up to date compared to available packages.

It will also update Git from 2.39.2 to 2.42.0

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -35,32 +69,39 @@ LABEL org.opencontainers.image.documentation="https://github.com/swissgrc/docker
...
+# Install Git 
+
+# renovate: datasource=repology depName=debian_12/libcurl3-gnutls versioning=loose
+ENV LIBCURL_VERSION=7.88.1-10+deb12u4
+
+# Install necessary dependencies
+RUN apt-get install -y --no-install-recommends libcurl3-gnutls=${LIBCURL_VERSION}
+
+# Copy Git
+COPY --from=build /usr/bin/git /usr/bin/git
```
Comment (Xzelsius) [inline Dockerfile:98]: What is the best way to know whether a new versions of Git adds new folders to copy?

**Review by Xzelsius — APPROVED** [review]

**Review by soebbel — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (pascalberger) [reply Dockerfile:98]: Beside the individual files to copy, there are also other aspects which are relevant, like what packages are available for building (e.g. for HTTPS support). I therefore would suggest to later add a integration test to the pipeline, which runs the container and tries to clone a repo (which is the main use case for git inside the container).

Comment (Xzelsius) [reply Dockerfile:98]: That would be great. Will you create an issue to track this?

Comment (pascalberger) [reply Dockerfile:98]: See #15

**Pull request merged**